csrc/optimization/pre_segmenter.cpp

```diff
@@ -9,10 +9,13 @@
 
 #include <optimization/add_axioms.h>
 #include <optimization/consecutive_cast.h>
+#include <optimization/remove_empty.h>
 
 namespace nvfuser::optimization {
 
 void PreSegmenter::runPass(Fusion* fusion) {
+  // Replace TensorViews with zero extent. Outputs and inputs may still be empty
+  OptimizationPass<RemoveEmptyPass>::runPass(fusion);
```

Comment:
I placed this pass first since I assumed we may want to do any DCE passes before other patterns are matched.